### PR TITLE
Removed filesystem and support saving with file name

### LINKED ISSUE #174: Question: Isn't it safe to remove filesystem?
All browser out there that has filesystem api also has download attribute?

### PR DESCRIPTION
**Pull request by jimmywarting**
closes #174
fixs #231

### CODE REVIEW

File: FileSaver.js [hunk 1]
```diff
@@ -135,20 +115,9 @@ var saveAs = saveAs || (function(view) {
...
-							return func.apply(this, arguments);
-						}
-					};
-				}
-				, create_if_not_found = {create: true, exclusive: false}
-				, slice
 			;
 			filesaver.readyState = filesaver.INIT;
-			if (!name) {
-				name = "download";
```
Comment (eligrey) [inline FileSaver.js]: This part is unrelated to filesystem support. It's simply for defaulting to a filename of `download` if no filename has been specified.

File: FileSaver.js [hunk 2]
```diff
@@ -161,93 +130,27 @@ var saveAs = saveAs || (function(view) {
...
 		};
 	}
 
-	FS_proto.abort = function() {
-		var filesaver = this;
-		filesaver.readyState = filesaver.DONE;
-		dispatch(filesaver, "abort");
-	};
+	FS_proto.abort = function(){};
 	FS_proto.readyState = FS_proto.INIT = 0;
```
Comment (eligrey) [inline FileSaver.js]: I never really documented filesaver.readyState, so I think it might be safe to remove this stuff (including the constants) as well. I don't think anyone is really using them.

Comment (jimmywarting) [inline FileSaver.js]: I don't believe so either. it also don't return the same thing when msSaveOrOpenBlob is being used so... And the progress/write event is a little bit confusing. you could expect it to do something when writing to the filesystem. but all it did was sending events when the sandboxed filesystem did stuff

You could get a progress event saying it's 100% done but what it really mean is that it is ready to start saving the file

So i think it's fundamental broken

the abort function was only useful for the sandboxed filesystem, so i made it to a noop function

**Comment by eligrey** [comment]
lgtm if you want to merge
